libs/core/langchain_core/messages/ai.py

```diff
@@ -29,6 +29,62 @@
 from langchain_core.utils.json import parse_partial_json
 
 
+class InputTokenDetails(TypedDict, total=False):
+    """Breakdown of input token counts.
+
+    Does *not* need to sum to full input token count. Does *not* need to have all keys.
+
+    Example:
+
+        .. code-block:: python
+
+            {
+                "audio": 10,
+                "cache_creation": 200,
+                "cache_read": 100,
+            }
+    """
+
+    audio: int
+    """Audio input tokens."""
+    cache_creation: int
```

Comment:
For OpenAI this will be `prompt_tokens  - prompt_tokens_details.cached_tokens`?

Reply:
i'm not actually sure if cached_tokens represents cache read or cache creation or a union of the two for oai, their docs aren't very clear

"cached_tokens: integer

Cached tokens present in the prompt."

Reply:
my understanding was its cache_read but will try to double check

Reply:
ok yea it represents cache_read. there isn't really a concept of cache_creation for oai since they do it automatically and don't charge extra for it, so think that would just be left out